src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -110,12 +120,7 @@ void LambdaFormInvokers::regenerate_holder_classes() {
   JavaValue result(T_OBJECT);
   JavaCalls::call_static(&result, cds_klass, method, signrs, list_lines, THREAD);
 
-  if (HAS_PENDING_EXCEPTION) {
-    log_info(cds)("%s: %s", THREAD->pending_exception()->klass()->external_name(),
-                            java_lang_String::as_utf8_string(java_lang_Throwable::message(THREAD->pending_exception())));
-    CLEAR_PENDING_EXCEPTION;
-    return;
-  }
+  HANDLE_IF_HAS_EXCEPTION;
```

Comment:
Instead of using a macro, I think the code can be more readable like this:

```
if (check_exception(THREAD)) {
    return;
}
```

and `check_exception` can contain the logic of `HANDLE_IF_HAS_EXCEPTION`.

Reply:
OK, I was think of that, if a function or macro and finally took macro.